tests/middleware/test_base.py

```diff
@@ -143,3 +143,49 @@ def homepage(request):
 def test_middleware_repr():
     middleware = Middleware(CustomMiddleware)
     assert repr(middleware) == "Middleware(CustomMiddleware)"
+
+
+async def numbers_stream(minimum, maximum):
```

Comment:
Should we remove these utility functions and write the app views inline without any dynamicism, one for each test case...?

- `/stream` - Streams successfully.
- `/stream-error-before-body` - Fail with an exception before reading the first body chunk.
- `/stream-error-in-body` - Fails with an exception while the body is being ready.

Right now the tests need some back and forth to read and understand, I understand the motivation for DRY for in this case I think some code repetition with a clear "here's plainly what we are testing against" might be better.

Reply:
That's fair. I pushed all the functionality into each tested handler.